xfields/ibs/_equilibrium.py

```diff
@@ -296,80 +330,84 @@ def compute_equilibrium_emittances_from_sr_and_ibs(
-        # convention is valid for arbitrary damping partition numbers and emittance_coupling_factor).
+    # First: if geometric emittances are provided we will just use those. If instead normalized
+    # emittances are provided, we convert those to geometric emittances
+    if nemitt_x is not None:
+        assert gemitt_x is None, "Cannot provide both 'gemitt_x' and 'nemitt_x'"
+        gemitt_x = nemitt_x / (twiss.beta0 * twiss.gamma0)
+    if nemitt_y is not None:
+        assert gemitt_y is None, "Cannot provide both 'gemitt_y' and 'nemitt_y'"
+        gemitt_y = nemitt_y / (twiss.beta0 * twiss.gamma0)
+    if nemitt_zeta is not None:
+        assert gemitt_zeta is None, "Cannot provide both 'gemitt_zeta' and 'nemitt_zeta'"
+        gemitt_zeta = nemitt_zeta / (twiss.beta0 * twiss.gamma0)
+    # ---------------------------------------------------------------------------------------------
+    # TODO: We can just check if ONE (or more) is None and say we take all 3 from TwissTable in this case -> any(emit is None for emit in (gemitt_x, gemitt_y, gemitt_zeta))
+    # Take default values from the TwissTable if no initial emittances are provided. In this case we
+    # need to renormalize with the emittance_constraint so we add a flag to know we need to do that
+    _renormalize_transverse_emittances = False
+    if gemitt_x is None and nemitt_x is None:
+        LOGGER.info("No horizontal emittance provided, taking SR equilibrium value from TwissTable.")
+        gemitt_x = twiss.eq_gemitt_x
+        _renormalize_transverse_emittances = True
+    if gemitt_y is None and nemitt_y is None:
+        LOGGER.info("No vertical emittance provided, taking SR equilibrium value from TwissTable.")
+        gemitt_y = twiss.eq_gemitt_y
+        _renormalize_transverse_emittances = True
+    if gemitt_zeta is None and nemitt_zeta is None:
+        LOGGER.info("No longitudinal emittance provided, taking SR equilibrium value from TwissTable.")
+        gemitt_zeta = twiss.eq_gemitt_zeta
+    # We need to also check for the case where use provided emittance and it is the twiss SR eq one
+    if gemitt_x == twiss.eq_gemitt_x or gemitt_y == twiss.eq_gemitt_y or gemitt_zeta == twiss.eq_gemitt_zeta:
+        _renormalize_transverse_emittances = True
+    # ---------------------------------------------------------------------------------------------
+    # By now we should have a value for geometric emittances in each plane. We assign them to new
+    # variables for clarity. These might be overwritten below in case we have to apply a constraint
+    starting_gemitt_x = gemitt_x
+    starting_gemitt_y = gemitt_y
+    starting_gemitt_zeta = gemitt_zeta
+    # ---------------------------------------------------------------------------------------------
+    # TODO: Decide with Seb if this is ok - we renormalize EVEN if just 1 value was taken from TwissTable SR eq -> not case if we do the TODO from above
```

Comment:
Potentially add a check that both transverse should be provided together or NOT provided together?

Reply:
In any case in documentation website example we provided 2 example cases to nudge (uwu) the users: one where nothing is provided (and we explicitely say and show it takes from TwissTable) and one where everything is provided.